app/controllers/course/discussion/posts_controller.rb

```diff
@@ -0,0 +1,106 @@
+class Course::Discussion::PostsController < Course::ComponentController
+  before_action :set_topic
```

Comment:
This abstract class only declares the interface. The base class for each specific condition is the one which actually initializes `@conditional`. And I think there are cases like `simple_form_for` requires the instance variable.